lib/telemetry_wrappers.ex

```diff
@@ -60,6 +60,45 @@ defmodule TelemetryWrappers do
     end
   end
 
+  @doc """
+  Defines a function that will have its execution time measured and sent as a telemetry event.
+  Also adds metadata to the metric.
+
+  As an example you can define the following module
+
+    defmodule TelemetryWrappers.Support.TestModule do
+      use TelemetryWrappers
+
+      deftimed timed_function_with_meta(a, b), [:a, :b], %{a: a} do
+        a + b
+      end
+    end
+
+  This will emit a `:telemetry` event `[:a, :b]` with the contents `%{call: timing}` and the metadata '%{a: a}'
+  where `timing` is the time the function took to execute in microseconds.
+
+  Note that type specs can still be defined for function defined with `deftimed` just as you would normally, e.g.
+
+      @spec timed_function(number(), number()) :: number()
+      deftimed timed_function_with_meta(a, b), [:a, :b], %{a: a} do
+        a + b
+      end
+  """
+  defmacro deftimed(function_name, metric_name, metadata, do: expr) do
```

Comment:
Wouldn't it be better to just default metadata to `%{}` and then re-use the same macro?